cpp/modmesh/inout/gmsh.cpp

```diff
@@ -69,7 +69,11 @@ Gmsh::Gmsh(const std::string & data)
 
 std::shared_ptr<StaticMesh> Gmsh::to_block()
 {
-    std::shared_ptr<StaticMesh> block = StaticMesh::construct(m_eldim.max(), m_nds.shape(0), 0, m_elems.size());
+    std::shared_ptr<StaticMesh> block = StaticMesh::construct(
+        m_eldim.max(),
+        static_cast<StaticMesh::uint_type>(m_nds.shape(0)),
```

Comment:
original is `size_t`, MSVS is not happy.

Reply:
What's the error/warning message from MSVC?

Reply:
key: ` conversion from 'size_t' to 'modmesh::StaticMesh::uint_type'`
```log
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.37.32822\include\xutility(254,82): warning C4267: 'argument': conversion from 'size_t' to 'modmesh::StaticMesh::uint_type', possible loss of data [D:\a\modmesh\modmesh\build\cpp\modmesh\modmesh_primary.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.37.32822\include\memory(2104,18): message : see reference to function template instantiation 'void std::_Construct_in_place<_Ty,unsigned int,size_t&,int,unsigned __int64,modmesh::StaticMesh::ctor_passkey>(_Ty &,unsigned int &&,size_t &,int &&,unsigned __int64 &&,modmesh::StaticMesh::ctor_passkey &&) noexcept(false)' being compiled [D:\a\modmesh\modmesh\build\cpp\modmesh\modmesh_primary.vcxproj]
          with
          [
              _Ty=modmesh::StaticMesh
          ]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.37.32822\include\memory(2779,46): message : see reference to function template instantiation 'std::_Ref_count_obj2<modmesh::StaticMesh>::_Ref_count_obj2<_Ty,size_t&,int,unsigned __int64,modmesh::StaticMesh::ctor_passkey>(_Ty &&,size_t &,int &&,unsigned __int64 &&,modmesh::StaticMesh::ctor_passkey &&)' being compiled [D:\a\modmesh\modmesh\build\cpp\modmesh\modmesh_primary.vcxproj]
          with
          [
              _Ty=unsigned int
          ]
D:\a\modmesh\modmesh\cpp\modmesh/mesh/StaticMesh.hpp(264,21): message : see reference to function template instantiation 'std::shared_ptr<_Ty> std::make_shared<modmesh::StaticMesh,unsigned int,size_t&,int,unsigned __int64,modmesh::StaticMesh::ctor_passkey>(unsigned int &&,size_t &,int &&,unsigned __int64 &&,modmesh::StaticMesh::ctor_passkey &&)' being compiled [D:\a\modmesh\modmesh\build\cpp\modmesh\modmesh_primary.vcxproj]
          with
          [
              _Ty=modmesh::StaticMesh
          ]
D:\a\modmesh\modmesh\cpp\modmesh\inout\gmsh.cpp(72,53): message : see reference to function template instantiation 'std::shared_ptr<_Ty> modmesh::StaticMesh::construct<unsigned int,size_t&,int,unsigned __int64>(unsigned int &&,size_t &,int &&,unsigned __int64 &&)' being compiled [D:\a\modmesh\modmesh\build\cpp\modmesh\modmesh_primary.vcxproj]
          with
          [
              _Ty=modmesh::StaticMesh
          ]
  ```

Reply:
What do you think if we make an overload of `StaticMesh::construct` to accept `size_t`?

Reply:
It seems unnecessary to have overloads for both size_t and uint32_t in the constructor. I would rather modify the constructor to accept either size_t or uint32_t.

Reply:
How can you modify the constructor to accept either `size_t` or `uint32_t` without overloading it?  I made a mistake to say overload of `StaticMesh::construct` which uses perfect forwarding.  I meant overload of `StaticMesh` constructor.

Reply:
I meant, do we need two `StaticMesh` constructors that accept both `size_t` and `uint32_t`? I am not sure the benefit of doing this. Then, if someone want to pass `int` to the constructor, shall we also have an overloading for `int`? I think we should just define a constructor, and users should do the casting themselves while using the constructor.

Reply:
Signed integer is a different story than different width of unsigned integer.

I'll take what you like.  Let's just cast at the caller.

Another option I was pondering is to use pragma to suppress the msvc warning.  It makes the code explicit that for most compilers the unsigned coercion is OK and only msvc doesn't like it.  But it is not obviously better than explicit casting.

Let me just merge.